### PR TITLE
[doc] improve `hello.rst' (#941)

### PR DESCRIPTION
**Pull request by TroyZhai**
Related issue = #941  (if any)

I went through 'hello.rst' and fixed some language issues. Most of them are just me being nitpicking, but hope that helps. 😃 

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

### CODE REVIEW

File: docs/hello.rst [hunk 1]
```diff
@@ -96,23 +95,23 @@ Taichi code can run on CPUs or GPUs. Initialize Taichi according to your hardwar
 
     (OK: supported; N/A: not available)
 
-    With ``arch=ti.gpu``, Taichi will try to run on CUDA.
-    If CUDA is not supported on your machine, Taichi will fall back to Metal or OpenGL.
-    If no GPU backend (CUDA, Metal, or OpenGL) is supported, Taichi will fall back to CPUs.
+    With ``arch=ti.gpu``, Taichi will try to run with CUDA.
+    If CUDA is not supported on your machine, Taichi will fall back on Metal or OpenGL.
```
Comment (archibate) [inline docs/hello.rst:99]: fall back on? Not sure if this is correct phase...sorry.

File: docs/hello.rst [hunk 2]
```diff
@@ -139,7 +138,7 @@ You can also define Taichi **functions** with ``ti.func``, which can be called a
```
Comment (archibate) [inline docs/hello.rst:138]: I found that `note`'s L129-138 seems overlaps L127? Could you help to remove that? Thx!

File: docs/hello.rst [hunk 3]
```diff
@@ -186,7 +185,7 @@ when used at the outermost scope. Range-for loops can be nested.
                 for i in range(10): # Serial :-(
                     ...
 
-**Struct-for loops** have are particularly useful when iterating over (sparse) tensor elements.
+**Struct-for loops** are particularly useful when iterating over (sparse) tensor elements.
```
Comment (archibate) [inline docs/hello.rst:188]: Nice catch! I guess the original isn't grammarly correct at all.

File: docs/hello.rst (hunk 1 above)
Comment (TroyZhai) [reply docs/hello.rst:99]: Well, I'm pretty sure. If you google 'fall back to' you will get 'fall back on', like this 
image

File: docs/hello.rst (hunk 2 above)
Comment (TroyZhai) [reply docs/hello.rst:138]: > I found that `note`'s L129-138 seems overlaps L127? Could you help to remove that? Thx!

Could you please elaborate on that? I don't really get it 😮

File: docs/hello.rst (hunk 1 above)
Comment (archibate) [reply docs/hello.rst:99]: Cool! Aplogize for my poor English..

File: docs/hello.rst (hunk 2 above)
Comment (archibate) [reply docs/hello.rst:138]: Oh, sorry, I mislooked, there's no duplicate at all..

File: docs/hello.rst [hunk 4]
```diff
@@ -51,20 +51,19 @@ Now you are ready to run the Taichi code below (``python3 fractal.py``) to compu
...
+Taichi is a domain-specific language (DSL) embedded in Python. To make Taichi as easy to use as a Python package, 
+we have done heavy engineering with this goal in mind - letting every Python programmer write Taichi codes with 
+minimal learning effort. You can even use your favorite Python package management system, Python IDEs and other 
+Python packages in conjunction with Taichi.
 
 Portability
 -----------
 
-Taichi code can run on CPUs or GPUs. Initialize Taichi according to your hardware platform:
+Taichi codes run on either CPUs or GPUs. Initialize Taichi according to your hardware platform as follows:
```
Comment (yuanming-hu) [inline docs/hello.rst:66]:
```suggestion
Taichi programs run on either CPUs or GPUs. Initialize Taichi according to your hardware platform as follows:
```

Sorry about being nitpicking... But *code* should probably be used as a mass noun... https://english.stackexchange.com/questions/20455/is-it-wrong-to-use-the-word-codes-in-a-programming-context

**Comment by yuanming-hu** [comment]
I'm merging this so that v0.6.4 reflects the changes.

File: docs/hello.rst (hunk 4 above)
Comment (TroyZhai) [reply docs/hello.rst]: > Sorry about being nitpicking... But _code_ should probably be used as a mass noun... 

You're right. 👍 
For future reference, code is uncountable for 代码 but countable for 准则. 

Sorry about that but there is another 'codes' at line 59 (after the merge).

File: docs/hello.rst (hunk 1 above)
Comment (TroyZhai) [reply docs/hello.rst:99]: No worries! I'm glad I'm helpful 😃

File: docs/hello.rst (hunk 4 above)
Comment (yuanming-hu) [reply docs/hello.rst]: Sorry about missing that one in Ln 59... I'll fix that in the next documentation update PR. Thanks for your inputs!!

**Pull request merged**